File: frontend/src/types/beerWiki.ts
/** 品类简要 */
export interface BeerStyleBrief {
  id: number;
  name: string;
  name_en: string | null;
  category: string;
  description: string | null;
  icon: string | null;
}

/** 品类详情（含雷达图数据） */
export interface BeerStyleDetail {
  id: number;
  name: string;
  name_en: string | null;
  category: string;
  description: string | null;
  origin_story: string | null;
  tasting_notes: string | null;
  ibu_min: number | null;
  ibu_max: number | null;
  srm_min: number | null;
  srm_max: number | null;
  abv_min: number | null;
  abv_max: number | null;
  icon: string | null;
}

/** 酒款简要 */
export interface BeerItemBrief {
  id: number;
  name: string;
  name_en: string | null;
  brewery: string | null;
  country: string | null;
  style_name: string | null;
  abv: number | null;
  ibu: number | null;
  image_url: string | null;
  is_classic: number;
}

/** 酒款详情 */
export interface BeerItemDetail {
  id: number;
  name: string;
  name_en: string | null;
  brewery: string | null;
  country: string | null;
  style_id: number | null;
  style_name: string | null;
  description: string | null;
  abv: number | null;
  ibu: number | null;
  og: number | null;
  image_url: string | null;
  flavor_tags: string | null;
  is_classic: number;
}

/** 搜索联想项 */
export interface BeerSuggestItem {
  type: "style" | "beer";
  id: number;
  name: string;
  subtitle: string | null;
}

/** 搜索结果 */
export interface BeerSearchResult {
  styles: BeerStyleBrief[];
  beers: BeerItemBrief[];
}

/** 品类+酒款 */
export interface StyleBeersResponse {
  style: BeerStyleBrief;
  beers: BeerItemBrief[];
  classic_beers: BeerItemBrief[];
}

/** 相关帖子（社区联动） */
export interface RelatedPost {
  id: number;
  title: string;
  content: string;
  like_count: number;
  comment_count: number;
  created_at: string | null;
  user_nickname: string | null;
  user_avatar: string | null;
  liked: boolean;
}

/** 酒款详情 + 社区联动 */
export interface BeerDetailWithPosts {
  beer: BeerItemDetail;
  related_posts: RelatedPost[];
}

File: src/schema/beer_wiki.py
from typing import Optional, List
from pydantic import BaseModel


# ── 风格品类 ──
class BeerStyleBrief(BaseModel):
    """品类简要（列表/金刚区用）"""
    id: int
    name: str
    name_en: Optional[str] = None
    category: str
    description: Optional[str] = None
    icon: Optional[str] = None

    class Config:
        from_attributes = True


class BeerStyleDetail(BaseModel):
    """品类百科详情"""
    id: int
    name: str
    name_en: Optional[str] = None
    category: str
    description: Optional[str] = None
    origin_story: Optional[str] = None
    tasting_notes: Optional[str] = None
    ibu_min: Optional[int] = None
    ibu_max: Optional[int] = None
    srm_min: Optional[int] = None
    srm_max: Optional[int] = None
    abv_min: Optional[float] = None
    abv_max: Optional[float] = None
    icon: Optional[str] = None

    class Config:
        from_attributes = True


# ── 酒款 ──
class BeerItemBrief(BaseModel):
    """酒款简要"""
    id: int
    name: str
    name_en: Optional[str] = None
    brewery: Optional[str] = None
    country: Optional[str] = None
    style_name: Optional[str] = None
    abv: Optional[float] = None
    ibu: Optional[int] = None
    image_url: Optional[str] = None
    is_classic: int = 0

    class Config:
        from_attributes = True


class BeerItemDetail(BaseModel):
    """酒款详情"""
    id: int
    name: str
    name_en: Optional[str] = None
    brewery: Optional[str] = None
    country: Optional[str] = None
    style_id: Optional[int] = None
    style_name: Optional[str] = None
    description: Optional[str] = None
    abv: Optional[float] = None
    ibu: Optional[int] = None
    og: Optional[float] = None
    image_url: Optional[str] = None
    flavor_tags: Optional[str] = None
    is_classic: int = 0

    class Config:
        from_attributes = True


# ── 搜索 ──
class BeerSuggestItem(BaseModel):
    """搜索联想单项"""
    type: str          # "style" | "beer"
    id: int
    name: str
    subtitle: Optional[str] = None   # 风格品类名 或 酒厂名


class BeerSuggestResponse(BaseModel):
    items: List[BeerSuggestItem]


class BeerSearchResult(BaseModel):
    """搜索结果"""
    styles: List[BeerStyleBrief] = []
    beers: List[BeerItemBrief] = []


# ── 品类下的酒款列表 ──
class StyleBeersResponse(BaseModel):
    style: BeerStyleBrief
    beers: List[BeerItemBrief] = []
    classic_beers: List[BeerItemBrief] = []   # 经典代表作
